railties/lib/rails/tasks/restart.rake

```diff
@@ -1,4 +1,7 @@
 desc "Restart app by touching tmp/restart.txt"
 task :restart do
+  unless File.directory?('tmp')
+    system 'mkdir tmp'
```

Comment:
It will be better to keep it all ruby and use [Dir::mkdir](http://ruby-doc.org/core-2.2.0/Dir.html#method-c-mkdir) to avoid various platform problems.